js/src/modal/modal.spec.js

```diff
@@ -602,6 +800,751 @@ describe('Modal', () => {
 
       modal.show()
     })
+
+    it('should not add compensation padding when body boundary width is less than window width with no vertical scrollbar before showing and no vertical scrollbar after showing', done => {
+      loadModalCss()
+      document.body.style.width = window.innerWidth - 1 + 'px'
+      document.body.style.height = '1px'
+      document.documentElement.style.paddingRight = null
+      unstyleFixture()
+
+      fixtureEl.innerHTML = getModalInnerHtml()
+
+      const modalEl = fixtureEl.querySelector('.modal')
+      const modal = new Modal(modalEl)
+
+      modalEl.addEventListener('show.bs.modal', e => {
+        expect(e).toBeDefined()
+      })
+
+      modalEl.addEventListener('shown.bs.modal', () => {
+        expect(modalEl.style.paddingRight).toEqual('')
+
+        done()
+      })
+
+      modal.show()
+    })
+
+    it('should not add compensation padding when body boundary width (including negative margin-left) is greater than window width with no vertical scrollbar before showing and no vertical scrollbar after showing', done => {
+      loadModalCss()
+      document.body.style.width = window.innerWidth + 101 + 'px'
+      document.body.style.height = '1px'
+      document.body.style.marginLeft = '-100px'
+      document.documentElement.style.paddingRight = null
+      unstyleFixture()
+
+      fixtureEl.innerHTML = getModalInnerHtml()
+
+      const modalEl = fixtureEl.querySelector('.modal')
+      const modal = new Modal(modalEl)
+
+      modalEl.addEventListener('show.bs.modal', e => {
```

Comment:
Ok it seems this `show` event handler is something you copy/pasted please remove it when it's not needed

Reply:
That's correct